custom_components/localtuya/coordinator.py

```diff
@@ -342,9 +345,14 @@ async def close(self):
             self._connect_task.cancel()
             await self._connect_task
             self._connect_task = None
-        if self._interface is not None:
-            await self._interface.close()
-            self._interface = None
+
+        # Close subdevices first, to prevent them try to reconnect
+        # after gateway disconnected.
+        subdevices = list(self.sub_devices.values())
+        for subdevice in subdevices:
+            await subdevice.close()
+
+        await self.abort_connect()
```

Comment:
Why do we need this? closing the gateway should be enough, why do we need "2" awaits for when calling "subdevice.close()" and also "gateway calls subdevice.close()"

Reply:
The current startup process initiates connections for (fake) gateways only. Each gateway initiates connections for its subdevices. It is logically correct, because subdevices share their `_interface` with the gateway.

It is also logical for a gateway to close its subdevices in its `close()`. If a gateway does not close its subdevices, but only aborts the connection, then `disconnected()` is called for each subdevice, and the subdevices start reconnect tasks. This loop first calls `close` for subdevices, and then they have `_is_closing` flag set, and don't try to reconnect after further gateway's `abort_connect()`.

The next logical step would be to close only directly connected devices (WiFi and (fake) gateways), like it is done in #363. Then each `close()` will be called only once.

Reply:
Let me fix my hard-drive within a few days when new one arrive, hope I can also restore old data, then do more tests combined with other PR to see what causes the issue. However I assume that the issue shows up if quick reloads happen.